### ISSUE TITLE
Move reviewer gesture detection to Javascript

### ISSUE THREAD

**Issue by BrayanDSO** [ankidroid/Anki-Android#14303]
In order to detect tapping elements like `<details>` and buttons, we can't use the Webview API because it only detects links, images and other unrelated stuff. Because of that, issues like #9500 occur.

So, an alternative is to move the gestures detection to Javascript and do the callbacks with JavascriptInterface

Probably some JS library should be used to do it, but I don't have any experience here, so I can't advise

**Comment by BrayanDSO** [comment]
Here's how AnkiMobile avoids triggering tap gestures:

https://docs.ankimobile.net/more.html#javascript

```js
function _isLink(e) {
    // we need to look at the source element and all parent elements
    // except the document
    let node = e.srcElement;
    while (node && node != document) {
        const res = node.nodeName == "A"
            || node.onclick
            || node.nodeName == "BUTTON"
            || node.nodeName == "VIDEO"
            || node.nodeName == "SUMMARY"
            || node.nodeName == "INPUT"
            || node.getAttribute("contentEditable");
        // INPUT / div contentEditable
        if (res) {
            return true;
        }
        if (node.classList && node.classList.contains("tappable")) {
            return true;
        }
        node = node.parentNode;
    }
    return false;
}
```

**Comment by BrayanDSO** [comment]
Another point is that some JS libraries can handle multi-finger touches, so maybe new gestures like two/three finger touch can be added too

**Comment by basvdelsen** [comment]
> Here's how AnkiMobile avoids triggering tap gestures:
> 
> https://docs.ankimobile.net/more.html#javascript
> 
> ```js
> function _isLink(e) {
>     // we need to look at the source element and all parent elements
>     // except the document
>     let node = e.srcElement;
>     while (node && node != document) {
>         const res = node.nodeName == "A"
>             || node.onclick
>             || node.nodeName == "BUTTON"
>             || node.nodeName == "VIDEO"
>             || node.nodeName == "SUMMARY"
>             || node.nodeName == "INPUT"
>             || node.getAttribute("contentEditable");
>         // INPUT / div contentEditable
>         if (res) {
>             return true;
>         }
>         if (node.classList && node.classList.contains("tappable")) {
>             return true;
>         }
>         node = node.parentNode;
>     }
>     return false;
> }
> ```

So adding the class "tappable" will also work for swipe gesture like in scrollable elements?

**Comment by BrayanDSO** [comment]
Swipe gestures wouldn't need that because they shouldn't be triggered if the user is scrolling or dragging something

**Comment by huandney** [comment]
Hi,
Is there something like `tappable` for ankidroid?

**Comment by BrayanDSO** [comment]
Currently, no

**Comment by BrayanDSO** [comment]
It's going to be a part of the new reviewer (#14302)

**Comment by huandney** [comment]
> Currently, no

Is there an intention to add `tappable` for AnkiDroid? 
I'm creating a multiple-choice card, and this would be much better than dealing with `<a href="#">`.

Thank you.

**Comment by BrayanDSO** [comment]
> Is there an intention to add tappable for AnkiDroid?

Eventually, yes. Together with #14302

**Comment by helitopia** [comment]
Hello, @BrayanDSO @david-allison 
I am willing to try to complete the task (actually I need this change for my ongoing Anki deck project). Are you looking for volunteers for this issue?

**Comment by david-allison** [comment]
I'll defer to @BrayanDSO on this one

**Comment by BrayanDSO** [comment]
> Are you looking for volunteers for this issue?

No. I've already solved most of that. It will be one of the next PRs for the new reviewer.
